sympy/matrices/matrices.py

```diff
@@ -379,7 +379,7 @@ def cofactor_matrix(self, method="berkowitz"):
         return self._new(self.rows, self.cols,
                          lambda i, j: self.cofactor(i, j, method))
 
-    def det(self, method="bareiss"):
+    def det(self, **kwargs):
```

Comment:
I think this should stay as it was. As currently implemented, Matrix.det method supports only one keyword argument, namely "method". When I type `help(Matrix.det)` I currently get `det(self, method='bareiss')` which tells me what keyword argument is supported, and what its default value is. Replacing this by `det(self, **kwargs)` is making this less clear. 

Other methods involved in this PR do not actually use the keyword arguments, they just pass them along, and that's fine. This is the one that uses them, and it should be specific about what is implemented. For example, suppose I type `M.det(metod='lu')`. With your change, there would be no error; the keyword argument 'metod" would be received and silently ignored. And I would never know that because of my typo, my program was using a wrong method for determinants. 

This is why the method needs explicit keyword arguments, so that an exception "unexpected keyword argument" will be raised in case of typos or other errors.